RPIPs/RPIP-53.md

```diff
@@ -0,0 +1,162 @@
+
+With that being said, if you have an RPL withdrawal address set and a primary withdrawal address different from your node at the time of `targetSlot`, there are some important implications that need to be considered:
+
+1. You will no longer be able to claim rewards via your node for that interval. Only your primary withdrawal address can claim your ETH rewards and only your RPL withdrawal address can claim your RPL rewards. In other words, this effectively "locks in" the rewards for an interval to your primary and RPL withdrawal addresses. They're going to be the only addresses that can claim the rewards for that interval, even if your withdrawal addresses change in the future or if you unset the RPL withdrawal address.
+2. Your RPL withdrawal address will not be able to use the "Claim and Stake" function introduced in Atlas to stake RPL on behalf of your node within a single transaction. It will only be able to claim its RPL.
+3. Setting the RPL withdrawal address to a node address may have odd, and sometimes not immediately obvious, consequences due to the way rewards are consolidated. **Be very careful if you intend to do this.**
+4. If you use these files to calculate your node's total earnings in a vacuum (not total *claimable* earnings, just total earnings produced in general), this will break your workflow. You'll have to calculate your RPL earnings out of band and look at the minipool performance files to determine your Smoothing Pool ETH earned. This is probably more of a dashboard concern than an individual node operator concern.
+
+As a note, the rationale for the **or** condition in the logic is purely designed to reduce unnecessary transactions. If nodes without a primary withdrawal address followed the same "breakout" rule that resulted in two sets of claimers, the RPL withdrawal address would have to claim for that node twice: once for itself (the RPL portion) and once for the ETH (the node portion). The node wouldn't be able to claim its own ETH portion since it still has an RPL withdrawal address set, and that's what the contracts key on when determining whether or not a node is eligible to claim - hence why this proposal exists in the first place. The easy way to mitigate this issue is just to set a primary withdrawal address to something other than the node; a practice that every node should be employing anyway if at all possible.
+
+
+## Reference Implementation
+
+I've implemented v10 into a Smart Node branch [here](https://github.com/rocket-pool/smartnode/tree/rewards_v10). For comparison, here are the differences between v8 and v10:
+
+- [For the vanilla (non-rolling) generator](https://www.diffchecker.com/tP55l6Iw/)
+- [For the rolling record generator](https://www.diffchecker.com/acLvVhpq/)
```

Comment:
I think these diffs might need to be inlined or made assets to avoid external dependencies.

Reply:
Ok, replaced diffchecker with some vimdiffs in the assets.